### PR TITLE
ci(unit): adding new action for running unit tests

### LINKED ISSUE #7: github action for unit tests
# Story
As a developer of Combo,
I want to place CI/CD in place for unit tests,
So that the code quality can be verified on each PR..

# Summary
The unit tests are currently already inside of the repo and runnable via `make test`. This task is as simple as getting the initial github action setup started and putting the unit tests in it.

# Acceptance Criteria
- Unit tests runnable inside of Github actions per PR.

### PR DESCRIPTION
**Pull request by tylerslaton**
# Summary
Getting the first action down for the CI, more to follow in other PRs.

# Additions
- Github action for running Unit Tests

Closes #7

### CODE REVIEW

File: .github/workflows/unit.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: ci
+on: [push]
```
Comment (timflannagan) [inline .github/workflows/unit.yaml:2]: This is needed right now, but in the future, we may want to restrict _which_ PRs get run based on what changes are included in the PR. A common example is a PR that only changes markdown files (i.e. a docs-only PR) that running unit tests against may be a waste of CI resources.

Comment (timflannagan) [reply .github/workflows/unit.yaml:2]: A concrete example that excludes certain paths: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/.github/workflows/unit.yml#L6-L8.

**Review by timflannagan — APPROVED** [review]

File: .github/workflows/unit.yaml (hunk 1 above)
Comment (tylerslaton) [reply .github/workflows/unit.yaml:2]: Hm, that's interesting. Looking into OLM, I noticed this:

```yaml
on:
  push:
    branches:
      - master
  pull_request:
    paths-ignore:
    - '**/*.md'
```

Its an easy enough fix, would there be a reason to not add this now?

Comment (tylerslaton) [reply .github/workflows/unit.yaml:2]: Nice! I'll go ahead and add that then.

**Comment by timflannagan** [comment]
LGTM

**Review by timflannagan — APPROVED** [review]

**Pull request merged**